### ISSUE TITLE
Test cmake and c interface with PRIMA_REAL_PRECISION being in [32, 64, 128], PRIMA_INTEGER_KIND being in [16, 32, 64], and PRIMA_DEBUGGING being in [0, 1]

### ISSUE THREAD

**Issue by zaikunzhang** [libprima/prima#129]
The Fortran code allows REAL(RP) to be different from 64-bit real, and INTEGER(IK) to the different from 32-bit integer. This can be done by setting PRIMA_REAL_PRECISION and PRIMA_INTEGER_KIND in fortran/common/ppf.h.

We should make sure that the cmake and c interface (or any other interface) still work if we vary these values from the default. 

In addition, PRIMA_DEBUGGING in ppf.h controls whether the Fortran code is in the debug mode, which enables checking of pre/postconditions in the Fortran code. This should also be tested. 

In addition, **PRIMA does not tolerate any warning**. This has been achieved by the Fortran code. The standard should be maintained by any other interface and implementation. 

@nbelakovski , we discussed this before. Could you have a look?

**Comment by nbelakovski** [comment]
Since the C interface has been written with `double` and `int` which are 8 and 4, respectively, on most platforms, does it make any sense to compile the C interface with `RP != 8`? Or would this effort include work to set up the C API in such a way that it can choose the precision of its floating point value based on `RP` or some other `#define`?

**Comment by zaikunzhang** [comment]
> Or would this effort include work to set up the C API in such a way that it can choose the precision of its floating point value based on RP or some other #define?

I don't think so. We only need to test. If someone wand RP different from 8, they can modify ppf.h manually.

**Comment by zaikunzhang** [comment]
> Or would this effort include work to set up the C API in such a way that it can choose the precision of its floating point value based on RP or some other #define?

The MATLAB interface provides two options `debug` and `precision` to end users. `debug` is a boolean, default to `false`; `precision` is a string and can be `"single"`, `"double"`, or `"quadruple"`, default to `"double"`. The interface will call an appropriate binary according to their values. There are 6 such binaries, compiled with 6 combinations of `PRIMA_DEBUGGING` and  `PRIMA_REAL_PRECISION`. 

Hopefully, other interfaces can do the same (one day).
